internal/usecases/pegin/get_pegin_report.go

```diff
@@ -0,0 +1,153 @@
+package pegin
+
+import (
+	"context"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/quote"
+	"github.com/rsksmart/liquidity-provider-server/internal/usecases"
+	"time"
+)
+
+type GetPeginReportUseCase struct {
+	peginQuoteRepository quote.PeginQuoteRepository
+}
+
+func NewGetPeginReportUseCase(
+	peginQuoteRepository quote.PeginQuoteRepository,
+) *GetPeginReportUseCase {
+	return &GetPeginReportUseCase{
+		peginQuoteRepository: peginQuoteRepository,
+	}
+}
+
+type GetPeginReportResult struct {
+	NumberOfQuotes     int
+	MinimumQuoteValue  *entities.Wei
+	MaximumQuoteValue  *entities.Wei
+	AverageQuoteValue  *entities.Wei
+	TotalFeesCollected *entities.Wei
+	AverageFeePerQuote *entities.Wei
+}
+
+func (useCase *GetPeginReportUseCase) Run(ctx context.Context, startDate time.Time, endDate time.Time) (GetPeginReportResult, error) {
+	var err error
+	var quotes []quote.PeginQuote
+	var minimumQuoteValue *entities.Wei
+	var maximumQuoteValue *entities.Wei
+	var averageQuoteValue *entities.Wei
+	var totalFeesCollected *entities.Wei
+	var averageFeePerQuote *entities.Wei
+
+	filter := quote.GetPeginQuotesByStateFilter{
+		States:    []quote.PeginState{quote.PeginStateRegisterPegInSucceeded},
+		StartDate: uint32(startDate.Unix()),
```

Comment:
I'd stick with the time.Time type until the actual query, is from the std lib so there is not problem on using it across the whole application